serverless/components
# State file based removals

## What has been implemented?

Implements removals based on information stored in the state file rather than the `serverless.yml` file.

Closes SC-182

## Steps to verify

Deploy the following service (`myFunction` depends on `myRole`):

```yml
type: test-application

components:
  myFunction:
    type: tests-integration-function-mock
    inputs:
      name: my-function
      role: ${myRole.id}
  myRole:
    type: tests-integration-iam-mock
    inputs:
      name: execution-role
      service: my.function.service
```

Now change the `serverless.yml` so that it looks like this (now `myRole` depends on `myFunction`). **Don't deploy it**:

```yml
type: test-application

components:
  myFunction:
    type: tests-integration-function-mock
    inputs:
      name: my-function
  myRole:
    type: tests-integration-iam-mock
    inputs:
      name: ${myFunction.name}-execution-role
      service: my.function.service
```

Run `components remove`. You should be able to remove the service.

## Todos:
